### DISCUSSION TITLE
Why does ZipRipper requires admin rights?

### DISCUSSION THREAD

**Discussion by nbourre** [illsk1lls/ZipRipper#38] [Q&A]
Why does ZipRipper requires admin rights? This is quite concerning.

**Comment by illsk1lls** [comment]
There are issues creating some temp files, and clearing old bitsadmin jobs out of programdata if multiple users run the script on the same system..

During testing it was found to be necessary..

It should not be concerning whatsoever, as you can see everything the script is doing, nothing is obfuscated, and I have gone out of my way to make it as readable as possible with labeling etc.. the only thing that the untrained eye might have an issue figuring out is the FOR loops with #'s and $'s and CMD guys can easily see they are just processing text files and matching lines... even the reason the menu PNG isnt B64 encoded directly into the file as a completely standalone CMD file is to avoid having any obfuscated data in the script..  So rest assured, unless 7zips public site, or JohnTheRipper/StrawberryPearl's github pages get compromised, then you have nothing to worry about.. All those links are clearly visible in the script..

**Reply by nbourre** [reply]
I've took a look at the script, but I stopped when it started to download files from places I didn't know.

Thanks

**Reply by illsk1lls** [reply]
That's the literal point of the script

Go ahead a familiarize yourself with the JohnTheRipper instructions for hashing Zip, 7zip, PDF, and RAR files manually, then you'll recognize what's going on...

Feel free to use John manually for the same effect.  Or you could use the official "Johnny" GUI version.

Either way, everything is visible for the user to decide for themselves, and nothing contained is malicious in any way shape or form.. ZipRipper is simply a shortcut for basic JtR usage for people who don't know ~ "All the places to get the right pre-reqs from, or what options to use"..

I use strawberry-perl portable so that you dont have to install perl on your system to hash 7z or RAR files, and I also skip installing cygwin on the system..  You will find the instructions provided around the web are actually telling you to download and install more pre-reqs than I'm using here with zipripper, what I am doing to get it to run on a windows system seems like the least invasive and bare minimal needed to complete the process as nothing is installed it is all temporary.. Check for yourself by doing it manually.

Regardless if you look further into the project, good luck with your archive
